test cases/rust/4 polyglot/meson.build

```diff
@@ -4,6 +4,46 @@ if host_machine.system() == 'darwin'
   error('MESON_SKIP_TEST: does not work right on macos, please fix!')
 endif
 
-l = shared_library('stuff', 'stuff.rs', rust_crate_type: 'cdylib', install : true)
-e = executable('prog', 'prog.c', link_with : l, install : true)
-test('polyglottest', e)
+extra_deps = [
+  dependency('threads'),
```

Comment:
Also `-ldl` apparently. Theoretically we can collect all these by passing `--print=native-static-libs` to `rustc` when linking the `staticlib`

Reply:
What's weird is it works fine locally. Wondering if rustc now add them by default? In any case, CI should be fixed now.